index.js

```diff
@@ -1,47 +1,53 @@
 'use strict';
 
-let AWSXRay = require('aws-xray-sdk');
-let url_lib = require('url');
-let https = AWSXRay.captureHTTPs(require('https'));
-let AWS = AWSXRay.captureAWS(require('aws-sdk'));
-let s3 = new AWS.S3();
+const AWSXRay = require('aws-xray-sdk');
+const urllib = require('url');
+const https = AWSXRay.captureHTTPs(require('https'));
+const AWS = AWSXRay.captureAWS(require('aws-sdk'));
+const s3 = new AWS.S3();
 
-let reverse = (s) => {
-  var o = '';
+exports.reverse = (s) => {
+  const trimmed = s.trim();
+  let o = '';
 
-  for (var i = s.length - 1; i >= 0; i--)
-    o += s[i];
+  for (let i = trimmed.length - 1; i >= 0; i--)
+    o += trimmed[i];
 
   return o;
 };
 
+exports.createS3UploadParams = (bucketname, data) => {
+  // The synctoken is unix epoch time, using a regex is faster than parsing JSON :)
+  const [ , regexToken ] = data.match(/syncToken\D+(\d+)/);
+  return {
+    Bucket: bucketname,
+    Key: `${exports.reverse(regexToken)}-ipranges.json`,
+    Body: data,
+  };
+};
+
 exports.handler = (event, context, callback) => {
   let message = event.Records[0].Sns.Message;
   if (typeof message === 'string') {
     message = JSON.parse(message);
   }
-  let url = message.url || "https://ip-ranges.amazonaws.com/ip-ranges.json";
+  const url = message.url || "https://ip-ranges.amazonaws.com/ip-ranges.json";
   console.log('Going to fetch ' + url);
-  let options = url_lib.parse(url);
+  const options = urllib.parse(url);
   https.get(options, function (res) {
     res.setEncoding('utf8');
-    var rawData = '';
+    let rawData = '';
     res.on('data', (chunk) => { rawData += chunk; });
     res.on('end', () => {
-      let regexToken = rawData.match(/syncToken[^0-9]+([0-9]+)/)[1];
       console.log(rawData);
-      var params = {
-        "Bucket": process.env.S3Bucket,
-        "Key": `${reverse(regexToken)}-ipranges.json`,
-        "Body": rawData
-      };
-      s3.upload(params, function(err, data) {
+      const params = exports.createS3UploadParams(process.env.S3Bucket, rawData);
```

Comment:
Codacy has a fix for the issue: [Replace `process.env.S3Bucket,·rawData` with `⏎········process.env.S3Bucket,⏎········rawData⏎······`](https://app.codacy.com/gh/dmgress/aws-ip-ranges/pullRequest?prid=9261990)
```suggestion
      const params = exports.createS3UploadParams(
```